cloudbridge/cloud/providers/gce/services.py

```diff
@@ -267,16 +269,84 @@ def current(self):
         return self.get(self.provider.region_name)
 
 
+class GCEImageService(BaseImageService):
+
+    def __init__(self, provider):
+        super(GCEImageService, self).__init__(provider)
+
+    def get(self, image_name):
+        """
+        Returns an Image given its name
+        """
+        try:
+            image = self.provider.gce_compute \
+                                  .images() \
+                                  .get(project=self.provider.project_name,
+                                       image=image_name) \
+                                  .execute()
+            if image:
+                return GCEMachineImage(self.provider, image)
+        except TypeError as type_error:
+            # The API will throw an TypeError, if parameter `image` does not
+            # match the pattern "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?".
+            print("TypeError: {0}".format(type_error))
+            return None
+        except googleapiclient.errors.HttpError as http_error:
+            print("googleapiclient.errors.HttpError: {0}".format(http_error))
+            return None
+
+    def find(self, name, limit=None, marker=None):
+        """
+        Searches for an image by a given list of attributes
+        """
+        filters = {'name': name}
+        try:
+            response = self.provider.gce_compute \
+                                    .images() \
+                                  .list(project=self.provider.project_name) \
+                                  .execute()
+        except googleapiclient.errors.HttpError as http_error:
+            print("googleapiclient.errors.HttpError: {0}".format(http_error))
+            return []
+        if 'items' not in response:
+            return []
+        images = [GCEMachineImage(self.provider, image) for image
+                  in response['items']
+                  if 'name' in image and image['name'] == filters['name']]
+        return ClientPagedResultList(self.provider, images,
+                                     limit=limit, marker=marker)
+
+    def list(self, limit=None, marker=None):
+        """
+        List all images.
+        """
+        try:
+            response = self.provider.gce_compute \
+                                  .images() \
+                                  .list(project=self.provider.project_name) \
```

Comment:
We have another GCE-discrepancy from AWS|OS here. The `project` argument filters images for the given project. So, as currently implemented, I just get an empty list since it only inspects the current project. If I set the value of `project` arg to `debian-cloud` for example, I get a list of images from that project. I would think that the `list` image (as well as `find` and `get`) should look at all public and private images by default. That's the behavior for AWS & OS. How do we achieve this for GCE? From the web console, I see a list of 18 images covering all the basic operating systems. How is that retrieved? Alternatively, how can one get a list of projects on GCE? I found `debian-cloud` in the docs, but which other ones exist? Should we compose a list of projects whose images we list/use?